### PR TITLE
PP-683: Sending large job scripts over TPP causes issues with TPP. Sh…

### PR DESCRIPTION
**Pull request by prakashcv13**
…ould use direct TCP connection in those cases.

#### Issue-ID

* **[PP-683](https://pbspro.atlassian.net/browse/PP-683)**

#### Problem description
* Sending large job scripts over TPP causes issues with TPP. Should use direct TCP connection in those cases.

#### Cause / Analysis
* Basically, TPP was never designed to be able to send very large files. To this extent to had earlier added code to send_job() such that on a job rerun, large spool files are not sent over TPP, rather fall back to the old method of forking the server and sending the job over a direct TCP connection.
Turns out, some sites submit huge job scripts. Job script is still sent over TPP which causes problem.

#### Solution description
* Thus the solution checks the length of the job script and decides to send over TPP or TCP (after a fork). We optimize the read of the script from the database (since we don't want to read multiple time to avoid a performance hit).
The fix is suggested, implemented and tested by @subhasisb while I have authored the PTL test cases.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [x] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [x] I have added documentation over **[here](https://pbspro.atlassian.net/wiki/display/PD/Interface+Change)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/server/svr_func.c [hunk 1]
```diff
@@ -6495,13 +6495,16 @@ extern char *path_spool;
 
 /*
  * @brief
- *  	Loads the jobscript associated to the job from the database
+ *  	Loads the job-script associated to the job from the database.
+ *  	It populates the ji_script field of the job as well as returns
+ *      a pointer to the script
  *
- * @param[in]	pj	-	Job pointer
+ * @param[in] pj - Job pointer
```
Comment (bhroam) [inline src/server/svr_func.c]: If you're loading the job script into pj->ji_script, this is also an out

Comment (prakashcv13) [inline src/server/svr_func.c]: Changed and added a description as well. Thanks!

**Comment by arungrover** [comment]
Change looks good to me

**Comment by prakashcv13** [comment]
Thank you @arungrover and @bhroam. I have squashed and rebased.. request you to take it forward.

**Pull request merged**